pennylane/qchem/vibrational/taylor_ham.py

```diff
@@ -0,0 +1,571 @@
+            for m2 in range(m1):
+                bosonized_qm2 = _position_to_boson(m2, "q")
+                for m3 in range(m2):
+                    bosonized_qm3 = _position_to_boson(m3, "q")
+                    for deg_idx, Qs in enumerate(degs_3d):
+                        q1deg, q2deg, q3deg = Qs[:3]
+                        coeff = taylor_3D[m1, m2, m3, deg_idx]
+                        bosonized_qm1_pow = bosonized_qm1**q1deg
+                        bosonized_qm2_pow = bosonized_qm2**q2deg
+                        bosonized_qm3_pow = bosonized_qm3**q3deg
+                        ordered_dict += (
+                            coeff * bosonized_qm1_pow * bosonized_qm2_pow * bosonized_qm3_pow
+                        ).normal_order()
+
+    return BoseSentence(ordered_dict).normal_order()
+
+
+def _taylor_kinetic(taylor_coeffs_array, freqs, is_loc=True, uloc=None):
+    """Build kinetic term of Taylor form bosonic observable from provided coefficients
+
+    Args:
+        taylor_coeffs_array (list(float)): the coeffs of the Taylor expansion
+        freqs (list(float)): the frequencies
+        is_loc (bool): Flag whether the vibrational modes are localized. Default is True.
+        uloc (list(float)): localization matrix indicating the relationship between original and
+            localized modes
+
+    Returns:
+        BoseSentence: kinetic term of the Taylor hamiltonian for given coeffs
+    """
+    taylor_1D = taylor_coeffs_array[0]
+    num_modes, _ = np.shape(taylor_1D)
+
+    if is_loc:
+        alphas_arr = np.einsum("ij,ik,j,k->jk", uloc, uloc, np.sqrt(freqs), np.sqrt(freqs))
+    else:
+        alphas_arr = np.zeros((num_modes, num_modes))
+        for m in range(num_modes):
+            alphas_arr[m, m] = freqs[m]
+
+    kin_ham = BoseSentence({})
+    for m1 in range(num_modes):
+        pm1 = _position_to_boson(m1, "p")
+        for m2 in range(num_modes):
+            pm2 = _position_to_boson(m2, "p")
+            kin_ham += (0.5 * alphas_arr[m1, m2]) * (pm1 * pm2).normal_order()
+
+    return kin_ham.normal_order()
+
+
+def _taylor_harmonic(taylor_coeffs_array, freqs):
+    """Build harmonic term of Taylor form bosonic observable from provided coefficients, see first
+    term of `Eq. 4 and Eq. 7 <https://arxiv.org/pdf/1703.09313>`_.
+
+    Args:
+        taylor_coeffs_array (list(float)): the coeffs of the Taylor expansion
+        freqs (list(float)): vibrational frequencies
+
+    Returns:
+        BoseSentence: harmonic term of the Taylor hamiltonian for given coeffs
+    """
+    taylor_1D = taylor_coeffs_array[0]
+    num_modes, _ = np.shape(taylor_1D)
+    harm_pot = BoseSentence({})
+    for mode in range(num_modes):
+        bosonized_qm2 = (
+            _position_to_boson(mode, "q") * _position_to_boson(mode, "q")
+        ).normal_order()
+        harm_pot += bosonized_qm2 * freqs[mode] * 0.5
+
+    return harm_pot.normal_order()
+
+
+def taylor_bosonic(taylor_coeffs_array, freqs, is_loc=True, uloc=None):
```

Comment:
```suggestion
def taylor_bosonic(taylor_coeffs, freqs, is_local=True, uloc=None):
```

Reply:
using `taylor_coeffs` is a bit tricky because we have a function also called `taylor_coeffs` (codefactor will complain).

Reply:
The please use `coeffs`.